### ISSUE TITLE
Feature: Support Cucumber usage reporter

### ISSUE THREAD

**Issue by adamwood-novafori** [vitalets/playwright-bdd#113]
**The problem**
I find it hard to find out if I have any unused steps

**A solution**
When running `bddgen export` it would be nice to see a count next to the step name to locate any unused ones

**Comment by Zialus** [comment]
This is available as the `usage` reporter of cucumber.js, but It seems that it isn't supported yet even after the recent addition of cucumber reports on playwright-bdd https://vitalets.github.io/playwright-bdd/#/reporters/cucumber

**Comment by vitalets** [comment]
Good point, we should definitely support [usage reporter](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md#usage). The reason why it is not implemented yet - it relies on step definitions. To have them in Playwright reporter output, playwright-bdd should re-read step definition files once again in the reporter process or pass them as a metadata. 

Another option as a lightweight solution - enhance `bddgen export` command to output unused steps, e.g. `bddgen export --unused-steps`. The output will not contain duration of each step, but the command will run very fast, because it does not  actually run tests.

**Comment by adamwood-novafori** [comment]
Maybe lightweight solution first and support usage reporter further down the line?

**Comment by vitalets** [comment]
Partially related to https://github.com/vitalets/playwright-bdd/issues/53

**Comment by vitalets** [comment]
Flag `--unused-steps` released in **v6.1.0**. Now you can run:
```
npx bddgen export --unused-steps
```
Output:
```
Using config: playwright.config.ts
List of unused steps (1):
* Then I see in title {string}
```
Currently unused steps are displayed without file location, will add it in the next release.

Keep this issue open for Cucumber usage reporter.

**Comment by harishh3492** [comment]
@vitalets I am trying to use the "bddgen export --unused-steps", 
* is there a way by which this can be restricted to check at a feature file level alone instead of the complete project? 
* Step definitions are not matching for 'And' keyword. I have a feature line with And(chained with Then), and the same has Then in the step def files, this is getting reported as unused in the report. 

playwright-bdd version: 8.2.0

**Comment by vitalets** [comment]
> is there a way by which this can be restricted to check at step defs used in particular feature file alone instead of the complete project?

There is no direct option for that, but you can use env variable to pass required feature and handle this variable in the Playwright config:

```ts
const testDir = defineBddConfig({
  features: process.env.FEATURE || 'features/*.feature',
  steps: 'features/steps/*.ts',
});
```
Then in the terminal:
```
FEATURE=features/my.feature npx bddgen export --unused-steps
```
You can restrict `steps` as well.

For the second point about matching for 'And' keyword, could you provide a reproduction code?

**Comment by harishh3492** [comment]
> 'And' keyword, could you provide a reproduction code?

@vitalets Apologies, my first question was misleading, there is no point in restricting at feature file level. The usage of Step to filter at step defs are certainly helpful. **Tried and works well**

The 'And' was a mistake from my end and the matching also works fine.

**Comment by vitalets** [comment]
No problem, thanks for confirming that it works!
